src/coreclr-debug/activate.ts

```diff
@@ -113,3 +111,39 @@ function showDotnetToolsWarning(message: string) : void
             });
     }
 }
+
+interface AdapterExecutableCommand {
+    command: string;
+}
+
+// The default extension manifest calls this command as the adapterExecutableCommand
+// If the debugger components have not finished downloading, the proxy displays an error message to the user
+// Else it will launch the debug adapter
+export async function registerAdapterExecutionCommand(channel: vscode.OutputChannel): Promise<AdapterExecutableCommand> {
+    let logger = new Logger(text => channel.append(text));
+    let util = new CoreClrDebugUtil(common.getExtensionPath(), logger);
+    let success = true;
+
+    if (!CoreClrDebugUtil.existsSync(util.installCompleteFilePath())) {
+        // our install.complete file does not exist yet, meaning we have not completed the installation. Try to figure out what if anything the package manager is doing
+        // the order in which files are dealt with is this:
+        // 1. install.Begin is created
+        // 2. install.Lock is created
+        // 3. install.Begin is deleted
+        // 4. install.complete is created
+
+        success = await completeDebuggerInstall(logger, channel);
+    }
+
+    if (success)   
+    {
+        // debugger has finished install and manifest has been rewritten, kick off our debugger process
+        return {
+            command: path.join(common.getExtensionPath(), ".debugger", "vsdbg-ui" + CoreClrDebugUtil.getPlatformExeExtension())
+        };
+    }
+    else
+    {
+        return null;
```

Comment:
What does VS Code do if we return null? Does it do something more sensible if we throw an error?

Reply:
At the moment it will error with the error message if we throw, or if it is invalid it will be ```it does not exist``` or ```cannot read property of 'command' null```.

This will be fixed in https://github.com/Microsoft/vscode/issues/33801

Reply:
As we discussed in person, I think all these error should be thrown, not logged.